spring-cloud-gateway-core/src/test/java/org/springframework/cloud/gateway/filter/factory/HystrixGatewayFilterFactoryTests.java

```diff
@@ -56,6 +59,22 @@ public void hystrixFilterTimesout() {
 				.isEqualTo(String.valueOf(HttpStatus.GATEWAY_TIMEOUT.value()));
 	}
 
+	@Test
+	public void hystrixFilterServiceUnavailable() {
+		HealthCountsStream.reset();
+		Hystrix.reset();
+		ConfigurationManager.getConfigInstance()
+				.setProperty("hystrix.command.failcmd.circuitBreaker.forceOpen", true);
+
+		testClient.get().uri("/delay/3").header("Host", "www.hystrixfailure.org")
+				.exchange().expectStatus().isEqualTo(HttpStatus.SERVICE_UNAVAILABLE);
+
+		HealthCountsStream.reset();
+		Hystrix.reset();
+		ConfigurationManager.getConfigInstance()
+				.setProperty("hystrix.command.failcmd.circuitBreaker.forceClosed", true);
```

Comment:
This should be `.setProperty("hystrix.command.failcmd.circuitBreaker.forceOpen", false);` AFAIK.
`forceOpen` takes precedence over `forceClosed` so your circuit will stay open.
It could potentially impact other tests if they are run in random order.

Reply:
Indeed, the precedence is mentioned in [Hystrix documentation](https://github.com/Netflix/Hystrix/wiki/Configuration#circuitbreakerforceclosed).
The commit [e435532](https://github.com/spring-cloud/spring-cloud-gateway/pull/1230/commits/e4355320cc3937e698f9e7de4843c1e1318bb327) caters for the proper handling of the circuit without affecting any subsequent tests.